clion/src/191/main/kotlin/org/rust/clion/profiler/RsSymbolSearcher.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.clion.profiler
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.Project
+import com.intellij.profiler.model.NativeCall
+import com.intellij.psi.search.GlobalSearchScope
+import org.rust.lang.core.psi.RsFunction
+import org.rust.lang.core.psi.ext.qualifiedName
+import org.rust.lang.core.stubs.index.RsNamedElementIndex
+import org.rust.openapiext.getElements
+
+class RsSymbolSearcher(private val project: Project) {
+    fun find(nativeCall: NativeCall): Array<RsFunction> {
+        val elements = mutableListOf<RsFunction>()
+        val searchScope = GlobalSearchScope.allScope(project)
+        val cargoTarget = extractPath(nativeCall.className)
+
+        ApplicationManager.getApplication().runReadAction {
+            val allDeclarations = getElements(RsNamedElementIndex.KEY, nativeCall.methodOrFunction, project, searchScope)
+                .filterIsInstance<RsFunction>()
+            val appropriateDeclarations = allDeclarations
+                .filter { it.qualifiedName?.substringBeforeLast("::") == cargoTarget }
+
+            if (appropriateDeclarations.isNotEmpty()) {
+                elements.addAll(appropriateDeclarations)
+            } else {
+                elements.addAll(allDeclarations)
+            }
+        }
+
+        return elements.toTypedArray()
+    }
+
+    /** `_<foo::bar123::foo_bar::Qux as baz>::qux` -> `foo::bar123::foo_bar` */
+    private fun extractPath(signature: String): String = signature
+        .dropWhile { !it.isLetter() }
+        .takeWhile { (it.isLetterOrDigit() && it.isLowerCase()) || it == '_' || it == ':' }
```

Comment:
Do I understand correctly that if module or crate name contains letter in upper case, it will work incorrectly?